packages/api/src/external/carequality/patient-external-data.ts

```diff
@@ -0,0 +1,74 @@
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import { cloneDeep } from "lodash";
+import { getPatientOrFail } from "../../command/medical/patient/get-patient";
+import { Patient } from "../../domain/medical/patient";
+import { PatientModel } from "../../models/medical/patient";
+import { executeOnDBTx } from "../../models/transaction-wrapper";
+import { PatientDataCarequality } from "./patient-shared";
+
+dayjs.extend(duration);
+
+export type PatientWithCQData = Patient & {
+  data: { externalData: { CAREQUALITY: PatientDataCarequality } };
+};
+
+/**
+ * Sets the Carequality (CQ) IDs on the patient.
+ *
+ * @param patientId The patient ID @ Metriport.
+ * @param cxId The customer ID @ Metriport.
+ * @param carequalityPatientId The patient ID @ cq gateway
+ * @param carequalityPatientSystemId The system ID @ cq gateway
+ * @returns
+ */
+export const setCarequalityId = async ({
+  patientId,
+  cxId,
+  carequalityPatientId,
+  carequalityPatientSystemId,
+}: {
+  patientId: string;
+  cxId: string;
+  carequalityPatientId: string;
+  carequalityPatientSystemId: string;
+}): Promise<Patient> => {
+  return executeOnDBTx(PatientModel.prototype, async transaction => {
+    const updatedPatient = await getPatientOrFail({
+      id: patientId,
+      cxId,
+      lock: true,
+      transaction,
+    });
+
+    const updatedData = cloneDeep(updatedPatient.data);
```

Comment:
can we raise a ticket to try this method in both places, then thoroughly test it, and prio accordingly instead?

we've had so much pain with these "sequelize isn't updating the data" scenarios that I don't have confidence this will work 100% of the time in every edge case

Reply:
https://github.com/metriport/metriport-internal/issues/1353